packages/nextjs/src/app-router/server/ClerkProvider.tsx

```diff
@@ -69,30 +71,36 @@ export async function ClerkProvider(
     </ClientClerkProvider>
   );
 
-  const shouldRunAsKeyless = !propsWithEnvs.publishableKey && canUseKeyless;
+  const runningWithClaimedKeys = propsWithEnvs.publishableKey === safeParseClerkFile()?.publishableKey;
+  const shouldRunAsKeyless = (!propsWithEnvs.publishableKey || runningWithClaimedKeys) && canUseKeyless;
 
   if (shouldRunAsKeyless) {
     // NOTE: Create or read keys on every render. Usually this means only on hard refresh or hard navigations.
     const newOrReadKeys = await import('../../server/keyless-node.js').then(mod => mod.createOrReadKeyless());
 
     if (newOrReadKeys) {
       const KeylessCookieSync = await import('../client/keyless-cookie-sync.js').then(mod => mod.KeylessCookieSync);
-      output = (
-        <KeylessCookieSync {...newOrReadKeys}>
-          <ClientClerkProvider
-            {...mergeNextClerkPropsWithEnv({
-              ...rest,
-              publishableKey: newOrReadKeys.publishableKey,
-              __internal_claimKeylessApplicationUrl: newOrReadKeys.claimUrl,
-              __internal_copyInstanceKeysUrl: newOrReadKeys.apiKeysUrl,
-            })}
-            nonce={await generateNonce()}
-            initialState={await generateStatePromise()}
-          >
-            {children}
-          </ClientClerkProvider>
-        </KeylessCookieSync>
+      const clientProvider = (
+        <ClientClerkProvider
+          {...mergeNextClerkPropsWithEnv({
+            ...rest,
+            publishableKey: newOrReadKeys.publishableKey,
+            __internal_claimKeylessApplicationUrl: newOrReadKeys.claimUrl,
+            __internal_copyInstanceKeysUrl: newOrReadKeys.apiKeysUrl,
+            __internal_keyless_dismissPrompt: runningWithClaimedKeys ? deleteKeylessAction : undefined,
```

Comment:
All of these props are related to keyless, but they use different naming schemes. Can we make them consistent? Proposal:

```suggestion
            __internal_keyless_claimApplicationUrl: newOrReadKeys.claimUrl,
            __internal_keyless_copyInstanceKeysUrl: newOrReadKeys.apiKeysUrl,
            __internal_keyless_dismissPrompt: runningWithClaimedKeys ? deleteKeylessAction : undefined,
```

Reply:
Yeap, seems like i never pushed this. Thanks for reminding me.